CalcMyTip/app/src/main/java/com/example/calcmytip/Bill.java

```diff
@@ -3,23 +3,39 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import android.content.Intent;
-import android.net.Uri;
 import android.os.Bundle;
 import android.view.View;
-import android.widget.MediaController;
-import android.widget.VideoView;
+import android.widget.EditText;
+import android.widget.TextView;
 
 public class Bill extends AppCompatActivity {
 
+    public static final String KEY_BILL = "billKey";
+    String tipStr;
+    TextView textTip;
+    EditText editTextBill;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_bill);
-
+        textTip = findViewById(R.id.textViewTip);
+        editTextBill = findViewById(R.id.textEditBill);
+        tipStr = getIntent().getStringExtra(RatingBase.KEY_TIP);
+        textTip.setText(tipStr);
     }
 
-    public void ToTipButton(View view) {
+    public void onCalculateBill(View view) {
+
+        String billStr = editTextBill.getText().toString();
+        int bill = Integer.parseInt(billStr);
+        int tip = Integer.parseInt(tipStr);
+
+        int total = (bill*(100+tip))/100;

```

Comment:
can you explain the formula in a comment?

Reply:
bill - is what the user entered.
tip is the calculated tip (from 0-18)
basic percentage formula....